sstables/sstables.cc

```diff
@@ -490,17 +490,17 @@ typename disk_set_of_tagged_union<TagType, Members...>::serdes disk_set_of_tagge
 
 template <typename TagType, typename... Members>
 future<>
-parse(sstable_version_types v, random_access_reader& in, disk_set_of_tagged_union<TagType, Members...>& s) {
+parse(schema_ptr schema, sstable_version_types v, random_access_reader& in, disk_set_of_tagged_union<TagType, Members...>& s) {
     using disk_set = disk_set_of_tagged_union<TagType, Members...>;
     using key_type = typename disk_set::key_type;
     using value_type = typename disk_set::value_type;
-    return do_with(0u, 0u, 0u, value_type{}, [&] (key_type& nr_elements, key_type& new_key, unsigned& new_size, value_type& new_value) {
-        return parse(v, in, nr_elements).then([&, v] {
+    return do_with(0u, 0u, 0u, value_type{}, [&, schema] (key_type& nr_elements, key_type& new_key, unsigned& new_size, value_type& new_value) {
```

Comment:
please think on all these schema captures if we can do a move capture. There are too many of these in this function for me to keep track ;-)